tensorflow/python/ops/init_ops.py

```diff
@@ -340,3 +341,47 @@ def __call__(self, shape, dtype=dtypes.float32, partition_info=None):
     """Generate a tensor used to initialize a variable."""
     return random_ops._random_walk(shape, self._nonlinearity, dtype,
                                    seed=self._seed)
+
+
+def orthogonal_initializer(gain=1.0, dtype=dtypes.float32, seed=None):
+  """Returns an initializer that generates an orthogonal rotation matrix.
+
+  Initialize a tensor using an orthogonal matrix.
+
+  Args:
+    gain: multiplicative factor to apply to the orthogonal matrix
+    dtype: The type of the output.
+    seed: None, a python integer to create a numpy random number generator or
+      an object that has a 'normal' member function
+
+  Returns:
+    An initializer that generates orthogonal tensors
+
+  Raises:
+    ValueError: if `dtype` is not a floating point type.
+  """
+  def _initializer(shape, dtype=_assert_float_dtype(dtype), partition_info=None):
+    if gain == 'relu':
+      _gain = math.sqrt(2)
+    else:
+      _gain = gain
+
+    # Use the default random number generator
+    if seed is None:
+      rng = np.random
+    # Use a specific seed
+    elif isinstance(seed, int):
+      rng = np.random.RandomState(seed)
+    else:
+      assert hasattr(seed, 'normal'), "rng must be None, an integer seed, or an object with a member function 'normal'"
+      rng = seed
+
+    # Flatten the input shape with the last dimension remaining its original shape so it works for conv2d
+    flat_shape = (np.prod(shape[:-1]), shape[-1])
+    a = rng.normal(0, 1, flat_shape)
+    u, _, v = np.linalg.svd(a, full_matrices=False)
```

Comment:
My biggest concern is why using np for this svd operation, instead of going through the tf.svd.